tests/unittests/Foundation/NSCharacterSetTests.m

```diff
@@ -152,71 +152,89 @@ void testCharacter(NSCharacterSet* charSet, NSString* testString) {
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet symbolCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet uppercaseLetterCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet whitespaceAndNewlineCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet whitespaceCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLFragmentAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLHostAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLPasswordAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLPathAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLQueryAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet URLUserAllowedCharacterSet];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet characterSetWithCharactersInString:@"ABCDEFG"];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
 
     set = [NSMutableCharacterSet characterSetWithRange:NSMakeRange(0, 32)];
-    ASSERT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+    EXPECT_TRUE([set isKindOfClass:[NSMutableCharacterSet class]]);
+}
+
+ARM_DISABLED_TEST(NSCharacterSet, ExceptionTests) {
+    NSMutableCharacterSet* set;
+    @try {
+        set = [NSMutableCharacterSet characterSetWithBitmapRepresentation:nil];
+        EXPECT_TRUE_MSG(false, "NSCharacterSet should have thrown invalid argument exception.");
+    } @catch (NSException* e) {
+        ASSERT_OBJCEQ(e.name, @"NSInvalidArgumentException");
+    }

```

Comment:
I'm not a fan of catching any throw. Sure it's expected that it's the correct throw but not guaranteed. It's more accurate to verify which exception was thrown.